packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -1122,6 +1140,37 @@ export default class Meeting extends StatelessWebexPlugin {
     });
   }
 
+  /**
+   * Notify any changes on the pstn devices
+   * @param {Object} payload
+   * @returns {undefined}
+   * @private
+   * @memberof Meeting
+   */
+  pstnUpdate(payload) {
+    if (this.locusInfo.self) {
+      const dialInPstnDevice = payload.newSelf?.pstnDevices.find((device) => device.url === this.dialInUrl);
+
+      if (dialInPstnDevice) {
+        this.isDialInDeviceActive = dialInPstnDevice.state === 'JOINED' || dialInPstnDevice.state === 'CONNECTED';
+        Trigger.trigger(
+          this,
+          {
+            file: 'meeting/index',
+            function: 'setUpLocusSelfListener'
+          },
+          EVENT_TRIGGERS.MEETING_SELF_PSTN_UPDATE,
+          {
+            dialIn: {
+              status: dialInPstnDevice.dialingStatus ?? dialInPstnDevice.state,
```

Comment:
why do we use dialingStatus here? it seems to be the only place in this whole PR. What's the relationship between dialInPstnDevice..dialingStatus and dialInPstnDevice..state?

Reply:
we get dialInPstnDevice from the device list, sent by the server. For the purposes of dialling-in, if dialingStatus exists, it'll only be "CONNECTED" (ie: the user has called in with their phone). If it doesn't exist, we'll use the state (which will only contain "JOINED" or "LEFT"). To clarify, when you're using your phone for audio, the state would be "JOINED" and dialingStatus would be "CONNECTED"; we're using the dialingStatus here to discriminate this from when you initially provision the device.


I think when you're dialling out, dialingStatus can be "RINGING" too. Will need to try out and see

Reply:
yeah, either remove it or change to `dialInPstnDevice.dialingStatus === 'CONNECTED'`, but I guess better to just remove it

Reply:
done. I decided to store the actual value. we will need it in leavePstn, when dial out is implemented